src/main/java/io/kestra/plugin/apify/task/AbstractGetDataset.java

```diff
@@ -0,0 +1,172 @@
+package io.kestra.plugin.apify.task;
+
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.models.property.Property;
+import io.kestra.core.runners.RunContext;
+import io.kestra.plugin.apify.ApifyConnection;
+import io.kestra.plugin.apify.ApifySortDirection;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import lombok.experimental.SuperBuilder;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+public abstract class AbstractGetDataset extends ApifyConnection {
+    @Schema(
+        title = "datasetId"
+    )
+    @NotNull
+    private Property<String> datasetId;
+
+
+    @Schema(
+        title = "Clean",
+        description = "If true then the task returns only non-empty items and skips hidden fields (i.e. fields starting with the # character). The default value is true."
+    )
+    private Property<Boolean> clean;
+
+    @Schema(
+        title = "Offset",
+        description = "Number of items that should be skipped at the start. The default value is 0."
+    )
+    private Property<Integer> offset;
+
+    @Schema(
+        title = "Limit",
+        description = "Maximum number of items to return. By default Limit value is set to 1000."
+    )
+    private Property<Integer> limit;
```

Comment:
For default values, you can do like this
```suggestion
    @Builder.default
    private Property<Integer> limit = Property.ofValue(1000);
```